doc/source/user/tutorial-svd.rst

```diff
@@ -0,0 +1,468 @@
+two-dimensional vector would be a list of two coordinates. In NumPy, however,
+a `dimension` is one of the axes in the array, so a two-dimensional vector
+is the equivalent of a matrix.
+
+First, let's check for the shape of the data in our array. Since this image is
+a two-dimensional set of data (the pixels in the image form a rectangle), we
+might expect a two-dimensional array to represent it (a matrix). However, using
+the ``shape`` property of this NumPy array gives us a different result:
+
+::
+
+    >>> img.shape
+    (768, 1024, 3)
+
+The output is a tuple, with three elements, which means that this is a
+three-dimensional array. In fact, since this is a color image, we have one set
+of pixel data per color (in this case, red, green and blue - RGB). You can see
+this by looking at the shape above: it indicates that we have 3 sets of 768x1024
+matrices.
+
+Furthermore, using the ``ndim`` property of this array, we can see that
+
+::
+
+    >>> img.ndim
+    3
+
+Each dimension is also referred to as an `axis` on NumPy. Because of how \
+color images work, the *first index in the 3rd axis* is the red pixel data for
+our image. We can access this by using the syntax
+
+::
+
+    >>> img[:, :, 0]
+    array([[121, 138, 153, ..., 119, 131, 139],
+           [ 89, 110, 130, ..., 118, 134, 146],
+           [ 73,  94, 115, ..., 117, 133, 144],
+           ...,
+           [ 87,  94, 107, ..., 120, 119, 119],
+           [ 85,  95, 112, ..., 121, 120, 120],
+           [ 85,  97, 111, ..., 120, 119, 118]], dtype=uint8)
+
+From the output above, we can see that every value in ``img[:,:,0]`` is an
+integer value between 0 and 255, representing the level of red in each
+corresponding image pixel.
+
+As expected, this is a 768x1024 matrix:
+
+::
+
+    >>> img[:, :, 0].shape
+    (768, 1024)
+
+Since we are going to perform linear algebra on this data, it might be more
+interesting to have real numbers between 0 and 1 in each entry of the matrices
+to represent the RGB values. We can do that by setting
+
+    >>> img_array = img/255
+
+You can check if this worked by doing some tests; for example, inquiring about
+maximum and minimum values for this array:
+
+::
+
+    >>> img_array.max(), img_array.min()
+    (1.0, 0.0)
+
+or checking the type of data in the array:
+
+::
+
+    >>> img_array.dtype
+    dtype('float64')
+
+Linear algebra on an axis
+-------------------------
+
+It is possible to use methods from linear algebra to approximate an existing set
+of data. Here, we will use the SVD (Singular Value Decomposition) to try to
+extract the most important information from this image.
+
+Keep in mind that most linear algebra operations (functions in the
+`numpy.linalg` module) can also be found in `scipy.linalg`. For more information
+on this, check the `numpy.linalg` reference.
+
+To proceed, import the linear algebra submodule from either NumPy or SciPy:
+
+    >>> import numpy.linalg as LA
```

Comment:
let's not do this, that's non-PEP 8 compliant old style. I know we may have this in other places still; we should nuke that.

`from scipy import linalg` is right for SciPy; for NumPy it could be simply `np.linalg.*` or `from numpy import linalg`. So the latter style would be better, so numpy/scipy are interchangeable.

Reply:
Agreed that we shouldn't do this, but I wasn't aware this went against PEP8 - can you point to the part it concerns?